src/register/data/optimizelyExperiment/useAutoGeneratedUsernameExperimentVariation.js

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useState } from 'react';
+
+import {
+  activateAutoGeneratedUsernameExperiment,
+  getAutoGeneratedUsernameExperimentVariation,
+  NOT_INITIALIZED,
+} from './helper';
+import { COMPLETE_STATE } from '../../../data/constants';
+
+/**
+ * This hook returns activates multi step registration experiment and returns the experiment
+ * variation for the user.
+ */
+const useAutoGeneratedUsernameExperimentVariation = (
+  registrationEmbedded,
+  tpaHint,
+  currentProvider,
+  thirdPartyAuthApiStatus,
+) => {
+  const [variation, setVariation] = useState('');
+  useEffect(() => {
+    if (registrationEmbedded || !!tpaHint || !!currentProvider
+        || thirdPartyAuthApiStatus !== COMPLETE_STATE) {
+      return variation;
+    }
+
+    const getVariation = () => {
+      const expVariation = getAutoGeneratedUsernameExperimentVariation();
+      if (expVariation) {
+        setVariation(expVariation);
+      } else {
+        // This is to handle the case when user dont get variation for some reason, the register page
+        // shows unlimited spinner.
+        setVariation(NOT_INITIALIZED);
+      }
+    };
+
+    activateAutoGeneratedUsernameExperiment();
```

Comment:
umm wouldn't this try to activate experiment everytime you switch between login/register?

maybe we can store variation fetched first time somewhere in redux and if we already have variation, we don't activate the experiment again.

Reply:
Done